File: training-lab/day-03/starter/transaction-alert.go
package main

type Transaction struct { ID string; AmountEUR int; Region string }
type Alert struct { Alert bool; Reason string; Severity string }

func alertForTransaction(transaction Transaction) Alert {
	if transaction.AmountEUR >= 9000 { return Alert{true, "amount exceeds the fictional review threshold", "high"} }
	return Alert{false, "within the fictional review threshold", "low"}
}

File: training-lab/day-03/starter/transaction-alert.ts
export type Transaction = { id: string; amountEur: number; region: string };
export type Alert = { alert: boolean; reason: string; severity: "high" | "low" };

export const alertForTransaction = (transaction: Transaction): Alert => transaction.amountEur >= 9000
  ? { alert: true, reason: "amount exceeds the fictional review threshold", severity: "high" }
  : { alert: false, reason: "within the fictional review threshold", severity: "low" };
